src/lucky/route_handler.cr

```diff
@@ -4,7 +4,21 @@ class Lucky::RouteHandler
   include HTTP::Handler
 
   def call(context : HTTP::Server::Context)
-    handler = Lucky.router.find_action(context.request)
+    original_path = context.request.path
+
+    # Extract format from URL path and strip it for route matching
+    if url_format = Lucky::MimeType.extract_format_from_path(original_path)
+      context._url_format = url_format
+      # Create a modified request with format-stripped path for route matching
+      path_without_format = original_path.sub(/\.[a-zA-Z0-9]+(?:\?.*)?$/, "")
```

Comment:
I think you have it covered already, but I couldn't find in the specs... does this handle if there's a domain in the query string? e.g. `/path/report.csv?affiliate=site.com/path.html`

Reply:
Good catch. I'm fairly certain that the current regex should match extensions in query strings (like .com in ?affiliate=site.com), but it won't cause issues because we're only recognizing registered formats. So unless the user went in and had a custom format that matched the TLD format, it should be fine. However, I think I can improve the regex a little bit.